external-crates/move/crates/move-vm-runtime/src/execution/values/values_impl.rs

```diff
@@ -2106,68 +2106,73 @@ impl GlobalValueImpl {
                 ))
             }
             Self::None => *self = Self::fresh(val)?,
-            Self::Deleted => *self = Self::cached(val, GlobalDataStatus::Dirty)?,
+            Self::Deleted { .. } => {
+                let Self::Deleted { fingerprint } = std::mem::replace(self, Self::None) else {
```

Comment:
Note: this is where the `fingerprint` inside the `Deleted` variant is used -- this is so we can track dirtying across a `move_from` and a `move_to`.